main.cpp

```diff
@@ -173,32 +173,27 @@ struct IntType
 };
 
 //point constructor implementations
-Point::Point(FloatType& a, FloatType& b) : x(a), y(b) {}
+Point::Point(FloatType& a, FloatType& b) : Point(static_cast<float>(a), static_cast<float>(b) ) { } 
 
-Point::Point(DoubleType& a, DoubleType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
+Point::Point(DoubleType& a, DoubleType& b) : Point ( static_cast<float>(a), static_cast<float>(b) ) { }
 
-Point::Point(IntType& a, IntType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
+
+Point::Point(IntType& a, IntType& b) : Point ( static_cast<float>(a), static_cast<float>(b) ) { }
 
 //point multiply UDT implementations
 Point& Point::multiply( FloatType& m )
 {
-    x *= m;
-    y *= m;
-    return *this;
+    return Point::multiply(static_cast<float>(m));
 }
 
 Point& Point::multiply( DoubleType& m )
 {
-    x *= static_cast<float>(m);
-    y *= static_cast<float>(m);
-    return *this;
+    return Point::multiply(static_cast<float>(m));
 }
 
 Point& Point::multiply( IntType& m )
 {
-    x *= static_cast<float>(m);
-    y *= static_cast<float>(m);
-    return *this;
+    return Point::multiply(static_cast<float>(m));
```

Comment:
you don't need to fully qualify the name.  it's not a base class function.  it's another member function
- [ ] revise